src/commands/purge.test.js

```diff
@@ -66,8 +66,8 @@ describe(`purge`, () => {
 
   beforeEach(() => {
     // Mock fastly dependency’s methods.
-    let purgeMock = function (serviceid, callback) {}
-    let softpurgeMock = function (serviceid, callback) {}
+    let purgeMock = function (serviceid, url, callback) {}
+    let softpurgeMock = function (serviceid, url, callback) {}
```

Comment:
These changes tell me that we're probably missing a test case in the `purge.test.js` file to validate that `purge()` stops without a url. I recommend pulling these two changes into a separate PR that also includes this new year case, with the added functionality to satisfy that test.